Neos.Flow/Classes/ObjectManagement/ObjectManagerInterface.php

```diff
@@ -43,21 +43,21 @@ public function getContext();
      * new keyword and Singleton objects should rather be injected by some type of
      * Dependency Injection.
      *
-     * @param string $objectName The name of the object to return an instance of
+     * @param string $id The name of the object to return an instance of
      * @param mixed[] ...$constructorArguments Any number of arguments that should be passed to the constructor of the object
      * @return object The object instance
      * @api
      */
-    public function get($objectName, ...$constructorArguments);
+    public function get(string $id);
```

Comment:
At least here the constructorArguments are needed. They are essential for the Flow Object manager and thus should be part of our interface. 

In the end the containerInterface is much less important for us than the objectManager.

Reply:
I'm not so sure about this. The argument was introduced in Feb only in https://github.com/neos/flow-development-collection/commit/d351daf324c11547d923abd83e6e57377288702e#diff-355665bb815e46477e5aff260fd21bfef08389279772e5d62330afdf0dd9620fR51 which hasn't even been released yet. Maybe @nezaniel can explain if this is really needed for the ESCR as is explained in the original PR https://github.com/neos/flow-development-collection/pull/2709 or if it was just one way to shut psalm up about the issue of calling this method with additional arguments at some places - in which case we are back to this exact "problem", which maybe just needs a baseline update instead of changing the signature?

Reply:
Then we should discuss this tomorrow with @nezaniel

Reply:
@albe the commit above added the argument to the interface but the objectManager had the additional constructor arguments for ages.

Reply:
Makes sense, yes. But not needed as a declared and typed argument IMHO. Of course it's nice that way, but since it works without just fine and the arguments are optional, I see no harm in undoing that specific change.

Why is that **needed** for the ESCR?

Reply:
This was not about phpstan, the ObjectManager did not properly implement its own interface, which threw errors on PHP8.1.
As long as this stays resolved, there should no issues with the CR

Reply:
> This was not about phpstan, the ObjectManager did not properly implement its own interface, which threw errors on PHP8.1.

It should work fine if the implementation adds an (variadic) argument:
https://3v4l.org/IdjQV#v8.1.4

Reply:
It does…

The current failures are caused Psalm (baseline update needed) and by the fact that [the used `behat/behat` version](https://github.com/neos/behat/blob/704ff2c4f3c5c5cfd930b39e785eb7b783db39b5/Resources/Private/Build/Behat/composer.json) 3.5 [requires `psr/container` as `~1.0`](https://packagist.org/packages/behat/behat#v3.10.0).

So we'd need to update `behat/behat`… but there is no newer version. 🤷‍♂️

Reply:
Tbh I don't really have an opinion on this. My change made the test suite run again with PHP8.1, which was all I needed for the ESCR. It might be very well related to behat or psalm, but anyway: Whatever way this is implemented, as long as I can run our behat test suite on PHP8.1, I'm fine.

Reply:
> The current failures are caused Psalm (baseline update needed) and by the fact that [the used `behat/behat` version](https://github.com/neos/behat/blob/704ff2c4f3c5c5cfd930b39e785eb7b783db39b5/Resources/Private/Build/Behat/composer.json) 3.5 [requires `psr/container` as `~1.0`](https://packagist.org/packages/behat/behat#v3.10.0).
> 
> So we'd need to update `behat/behat`… but there is no newer version. 🤷‍♂️

😭 okay, so guess we're blocked on this by third party for now then - not sure we can make this work with `psr/container ^1.0 || ^2.0`

Reply:
Hope exists: https://github.com/Behat/Behat/pull/1380

Reply:
:pray: but I doubt this will happen before we tag the Flow/Neos release - so something for 8.1

Reply:
https://github.com/Behat/Behat/pull/1380 is merged, https://github.com/Behat/Behat/pull/1386 closed. 🚀

Reply:
I reverted the change to the `get(…)` signature, I feel our interface should match our implementation…